addons/account/report/account_tax_details_report.py

```diff
@@ -0,0 +1,385 @@
+        readonly=True,
+        currency_field='currency_id',
+    )
+    company_id = fields.Many2one(
+        comodel_name='res.company',
+        string="Company",
+    )
+    company_currency_id = fields.Many2one(
+        comodel_name='res.currency',
+    )
+    currency_id = fields.Many2one(
+        comodel_name='res.currency',
+    )
+
+    def init(self):
+        self._cr.execute('''
+            CREATE OR REPLACE VIEW account_tax_details_report AS (
+
+                WITH tax_ids_m2m AS (
+
+                    SELECT
+                        tax_rel.account_move_line_id AS id,
+                        ARRAY_AGG(tax_rel.account_tax_id ORDER BY tax.sequence, tax.id) AS tax_ids
+                    FROM account_move_line_account_tax_rel tax_rel
+                    JOIN account_move_line base_line ON base_line.id = tax_rel.account_move_line_id
+                    JOIN account_tax tax ON tax.id = tax_rel.account_tax_id
+                    WHERE tax.amount_type != 'group'
+                        AND (tax.is_base_affected OR base_line.tax_repartition_line_id IS NOT NULL)
+                    GROUP BY tax_rel.account_move_line_id
+
+                    UNION ALL
+
+                    SELECT
+                        tax_rel.account_move_line_id AS id,
+                        ARRAY_AGG(group_tax_rel.child_tax ORDER BY tax.sequence, tax.id) AS tax_ids
+                    FROM account_move_line_account_tax_rel tax_rel
+                    JOIN account_tax tax ON tax.id = tax_rel.account_tax_id
+                    JOIN account_move_line base_line ON base_line.id = tax_rel.account_move_line_id
+                    JOIN account_tax_filiation_rel group_tax_rel ON group_tax_rel.parent_tax = tax.id
+                    WHERE tax.amount_type = 'group'
+                        AND (tax.is_base_affected OR base_line.tax_repartition_line_id IS NOT NULL)
+                    GROUP BY tax_rel.account_move_line_id
+
+                ),
+
+                base_tax_matching_no_tax_line_as_base AS (
+
+                    SELECT
+                        tax_line.id AS tax_line_id,
+                        base_line.id AS base_line_id,
+
+                        base_line.balance AS base_amount,
+                        base_line.amount_currency AS base_amount_currency
+
+                    FROM account_move_line tax_line
+                    JOIN account_tax_repartition_line tax_rep ON
+                        tax_rep.id = tax_line.tax_repartition_line_id
+                    JOIN account_tax tax ON
+                        tax.id = tax_line.tax_line_id
+                    JOIN res_currency curr ON
+                        curr.id = tax_line.currency_id
+                    JOIN res_currency comp_curr ON
+                        comp_curr.id = tax_line.company_currency_id
+                    JOIN account_move_line_account_tax_rel tax_rel ON
+                        tax_rel.account_tax_id = COALESCE(tax_line.group_tax_id, tax_line.tax_line_id)
```

Comment:
why do you join on tax_rel AND on tax_ids_m2m ?

Reply:
tax_ids_m2m is flattened taxes without the ones having is_base_affected set to FALSE

Here, I joining the originator m2m because I want to match on "group of taxes" if there some